### ISSUE TITLE
Connect to oplog and database with different user authentication and URL

### ISSUE THREAD

**Issue by bubuzzz** [mongodb-labs/mongo-connector#723]
I successfully setup elasticsearch with Mongo on my local. However, when we moved to mlab account, we encounter this exception 

`OperationFailure: not authorized on cba-demo to execute command { listCollections: 1, cursor: {} } `

I believe this is because we are using mlab shared plan (https://mlab.com/plans/), so the user we created for admin database does not have the permission across all other databases in order to query the collections of the main database. From the wiki page of mongo-connector, we can only pass in 1 single url and user authentication. Is there a way to overcome this issue ? This is the command i used to test with our mlab account 

`sudo mongo-connector -m "mongodb://user:pwd@xxxxxxx.mlab.com:11111,yyyyyyy-a1.mlab.com:22222/local?authSource=admin" -t localhost:9200  -d elastic2_doc_manager -c ./config.json`

**Comment by jayminkapish** [comment]
I was able to resolve this by reaching out to mlab support and ask to
assign oplog user readAnyDatabase role and use only this oplog user in my
config, leaving admin user block commented/ignored.



On Jul 31, 2017, at 8:06 PM, Thai Tran <notifications@github.com> wrote:

I successfully setup elasticsearch with Mongo on my local. However, when we
moved to mlab account, we encounter this exception

OperationFailure: not authorized on cba-demo to execute command {
listCollections: 1, cursor: {} }

I believe this is because we are using mlab shared plan (
https://mlab.com/plans/), so the user we created for admin database does
not have the permission across all other databases in order to query the
collections of the main database. From the wiki page of mongo-connector, we
can only pass in 1 single url and user authentication. Is there a way to
overcome this issue ? This is the command i used to test with our mlab
account

sudo mongo-connector -m "mongodb://user:pwd@xxxxxxx.mlab.com:11111,
yyyyyyy-a1.mlab.com:22222/local?authSource=admin" -t localhost:9200 -d
elastic2_doc_manager -c ./config.json

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/mongodb-labs/mongo-connector/issues/723>, or mute the
thread

**Comment by bubuzzz** [comment]
@jayminkapish  thank a lot, i will do so as well now. However, mongo-connector should be able to get both urls (like Meteor framework) so this one does not occur by default

**Comment by jayminkapish** [comment]
Yeah! Last I remember I was stuck at python-mongo remembering connection
and it wasn't easy to get around that other than commenting out some
caching code in python-mongo to test this. I am happy I reached out to mlab
support, they were very supportive and with this solution my config is
simple and only one user to manage.

Good luck!

On Jul 31, 2017, at 9:26 PM, Thai Tran <notifications@github.com> wrote:

@jayminkapish <https://github.com/jayminkapish> thank a lot, i will do so
as well now. However, mongo-connector should be able to get both urls (like
Meteor framework) so this one does not occur by default

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/mongodb-labs/mongo-connector/issues/723#issuecomment-319241371>,
or mute the thread
